src/qibo/models/dbi/double_bracket.py

```diff
@@ -114,47 +151,31 @@ def backend(self):
         """Get Hamiltonian's backend."""
         return self.h0.backend
 
-    def hyperopt_step(
+    def least_squares(self, D: np.array):
+        """Least squares cost function."""
+        H = self.h.matrix
+        return -np.real(
+            np.trace(H @ D) - 0.5 * (np.linalg.norm(H) ** 2 + np.linalg.norm(D) ** 2)
+        )
```

Comment:
Although we don't test on GPU in GitHub all DBI features should be GPU compatible (qibojit).
Here are the instructions on how to install [qibojit](https://qibo.science/qibo/stable/getting-started/installation.html#installing-qibojit).
After having installed qibojit (cupy) properly by running tests locally you should be able to see if tests are passing or not also for GPU.

Reply:
This is expected, here is a possible blueprint

https://github.com/qiboteam/qibo/blob/d0c0ba2c67350b634b7f419f640dbcea9cf732a1/src/qibo/models/dbi/double_bracket.py#L222-L223

Reply:
Basically the idea is that if you perform operation with numpy you should make sure that you are playing with a numpy element (which explains `backend.to_numpy`). Once you are finish with the numpy manipulations make sure to call `backend.cast` to convert back to whatever backend you were using in the first place.